base_tier_validation/wizard/comment_wizard.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class CommentWizard(models.TransientModel):
+    _name = 'comment.wizard'
+    _description = 'Comment Wizard'
+
+    validate_reject = fields.Char()
+    res_model = fields.Char()
+    view_id = fields.Many2one(
+        comodel_name='ir.ui.view',
+    )
+    definition_ids = fields.Many2many(
+        comodel_name='tier.definition',
+    )
+    comment = fields.Char(
+        required=True,
+    )
+
+    @api.multi
+    def add_comment(self):
+        self.ensure_one()
+        action = self.env['ir.model.data'].search([
+            ('model', '=', self.res_model),
+            ('res_id', '=', self.view_id.id),
+        ], limit=1)
+        act = self.env.ref(action.module + '.' + action.name)
```

Comment:
Again, if I am not wrong (correct me if I am wrong, coz I am very confused now... O_0), how come it has external id.
```suggestion
        act = self.env.ref(action.module + '.' + action.name)
        rec = self.env[self.res_model].browse(self.res_id)  # use rec instead of act.
```
I think we don't need to use self.env['ir.model.data'].search() at all.

Reply:
Edited